src/cli/java/org/commcare/util/screen/QueryScreen.java

```diff
@@ -196,17 +195,19 @@ public void answerPrompts(Hashtable<String, String> answers) {
                 Vector<SelectChoice> selectChoices = queryPrompt.getItemsetBinding().getChoices();
                 String[] indicesOfSelectedChoices = RemoteQuerySessionManager.extractMultipleChoices(answer);
                 ArrayList<String> selectedChoices = new ArrayList<>(indicesOfSelectedChoices.length);
-                for (int i = 0; i < indicesOfSelectedChoices.length; i++) {
-                    if (indicesOfSelectedChoices[i].isEmpty()) {
-                        selectedChoices.add("");
-                    } else {
-                        int choiceIndex = Integer.parseInt(indicesOfSelectedChoices[i]);
-                        if (choiceIndex < selectChoices.size() && choiceIndex > -1) {
-                            selectedChoices.add(selectChoices.get(choiceIndex).getValue());
+                if (!selectValuesByKeys) {
```

Comment:
seems like the whole outer `if` is only relevant if `!selectValuesByKeys`, is it possible to expand the scope of this if statement to include more?

Certainly the `selectedChoices` declaration can be moved in as well as `indicesOfSelectedChoices` and I think `selectChoices`. Not sure about `remoteQuerySessionManager.populateItemSetChoices(queryPrompt)`. Since that seems like it has side effects.

Reply:
Good tip - https://github.com/dimagi/commcare-core/pull/1261/commits/055af7bf4694dc077c9ff8f2c6a360e2821c1f43 , `remoteQuerySessionManager.populateItemSetChoices(queryPrompt)` is still required for both old and new formats so kept it outside.